src/Resolvers/Collections.ts

```diff
@@ -75,6 +75,39 @@ export class CollectionsResolver {
     return await this.repository.findOne({ slug })
   }
 
+  @FieldResolver(type => [Collection])
+  async relatedCollections(
+    @Root() collection: Collection
+  ): Promise<Collection[]> {
+    const isArtistBased =
+      collection.query.artist_ids && collection.query.artist_ids.length > 0
+
+    if (isArtistBased) {
+      const artistResults = await this.repository.find({
+        where: { "query.artist_ids": { $in: collection.query.artist_ids } },
+      })
+
+      const relatedArtistCollections = reject(artistResults, {
+        id: collection.id,
+      })
+
+      if (relatedArtistCollections.length > 4) {
+        return relatedArtistCollections
+      }
+    }
+
+    const relatedCategoryResults = await this.repository.find({
+      where: {
+        category: { $in: [collection.category] },
+        show_on_editorial: true,
```

Comment:
I also added a filter for `show_on_editorial` which is also mentioned in the requirements.